srtcore/common.cpp

```diff
@@ -566,33 +567,32 @@ bool CIPAddress::ipcmp(const sockaddr* addr1, const sockaddr* addr2, int ver)
    return false;
 }
 
-void CIPAddress::ntop(const sockaddr* addr, uint32_t ip[4], int ver)
+void CIPAddress::ntop(const sockaddr_any& addr, uint32_t ip[4])
 {
-   if (AF_INET == ver)
-   {
-      sockaddr_in* a = (sockaddr_in*)addr;
-      ip[0] = a->sin_addr.s_addr;
-   }
-   else
-   {
-      sockaddr_in6* a = (sockaddr_in6*)addr;
+    if (addr.family() == AF_INET)
+    {
+        ip[0] = addr.sin.sin_addr.s_addr;
```

Comment:
You can only see a change here because the code was reformatted. There's no reason to change the existing code here.

Actually this function is used exclusively to initialize a special field in the `CHandShake` object (both 2 places in the code, in core.cpp), just after creation, and the constructor of this type (refer to `handshake.cpp`) clears out this field. I'd agree that this deserves some refactoring, but it should start with moving this "ntop" function to be a method of `CHandShake`.